src/test/java/org/biouno/unochoice/issue72105/TestJobDslWithScriptlerScriptWithParameters.java

```diff
@@ -0,0 +1,141 @@
+     *
+     * @throws Exception I/O?
+     */
+    @Before
+    public void setUp() throws Exception {
+        ScriptApproval.get().preapprove(GROOVY_SCRIPT, new GroovyLanguage());
+        // scriptler setup
+        scriptler = j.getInstance().getExtensionList(ScriptlerManagement.class).get(0);
+        ScriptlerHelper scriptlerHelper = new ScriptlerHelper(scriptler);
+        scriptFile = Files.createTempFile("uno-choice", "Cascade.groovy").toFile();
+        FileUtils.writeStringToFile(scriptFile, GROOVY_SCRIPT, Charset.defaultCharset(), false);
+        FileItem fi = new FileParameterValue.FileItemImpl(scriptFile);
+        scriptlerHelper.saveScript(fi, true, "Cascade.groovy");
+
+        scriptler.getConfiguration()
+                .getScriptById("Cascade.groovy")
+                .setParameters(Arrays.asList(
+                        new Parameter("P1", ""),
+                        new Parameter("P2", "")));
+    }
+
+    @Test
+    @LocalData("test")
+    public void test() throws Exception {
+        assertNull(j.jenkins.getItem("scriptler-test"));
+
+        final FreeStyleProject job = (FreeStyleProject) j.jenkins.getItem("job");
+        j.buildAndAssertStatus(Result.UNSTABLE, job);
+
+        final WorkflowJob jobDSLCreated = (WorkflowJob) j.jenkins.getItem("scriptler-test");
+        assertNotNull(jobDSLCreated);
+
+        // This will be set to true if we are able to confirm P1 and P2 were persisted with the JobDSL created job
+        boolean parametersDefined = false;
+        loop1:
+        for (Map.Entry<JobPropertyDescriptor, JobProperty<? super WorkflowJob>> entry : jobDSLCreated.getProperties()
+                .entrySet()) {
+            JobProperty<? super WorkflowJob> jobProperty = entry.getValue();
+            if (jobProperty instanceof ParametersDefinitionProperty) {
+                ParametersDefinitionProperty paramDef = (ParametersDefinitionProperty) jobProperty;
+                List<ParameterDefinition> parameters = paramDef.getParameterDefinitions();
+                for (ParameterDefinition parameter : parameters) {
+                    if (PARAMETER_WITH_SCRIPTLER_AND_PARAMETERS.equals(parameter.getName()) &&
+                            parameter instanceof CascadeChoiceParameter) {
+                        final CascadeChoiceParameter p3 = (CascadeChoiceParameter) parameter;
+                        ScriptlerScript script = (ScriptlerScript) p3.getScript();
+                        assertEquals(2, script.getParameters().size());
+                        assertTrue(script.getParameters().containsKey("P1"));
+                        assertTrue(script.getParameters().containsKey("P2"));
+                        parametersDefined = true;
+                        break loop1;
+                    }
+                }
+            }
+        }
+
+        assertTrue("The P3 parameter MUST have a scriptler parameter with two parameters (persisted when saving it). Regression detected!", parametersDefined);
```

Comment:
@limejuny , this is the test that I think confirms your change works and prevents us from breaking the feature/behavior of the plug-in again.

Basically, it's a normal JUnit test, annotated with `@Test` but also with `@LocalData`. This last annotation instructions Jenkins test harness to use a `config.xml` that I saved after reproducing your setup (I just copied a file like `/work/jobs/job/config.xml` into the test resources correct location).

This means that when the test is started, because it has the `@Rule JenkinsRule` object, a Jenkins instance is created for this test, and the same `job` you described in the issue is created by Jenkins for test. Then this test above was quickly written (could be simpler I think, but strapped for time now) just checks the Job DSL creates a new job, and finally confirms that the `P3` parameter contains the scriptler script with the parameters persisted.

I executed this with the IDE, while I still had the breakpoint in the line you changed, and that was triggered as expected.